src/native/managed/cdacreader/src/Data/Module.cs

```diff
@@ -0,0 +1,96 @@
+
+        FieldDefToDescMap = target.ReadPointer(address + (ulong)type.Fields[nameof(FieldDefToDescMap)].Offset);
+        ManifestModuleReferencesMap = target.ReadPointer(address + (ulong)type.Fields[nameof(ManifestModuleReferencesMap)].Offset);
+        MemberRefToDescMap = target.ReadPointer(address + (ulong)type.Fields[nameof(MemberRefToDescMap)].Offset);
+        MethodDefToDescMap = target.ReadPointer(address + (ulong)type.Fields[nameof(MethodDefToDescMap)].Offset);
+        TypeDefToMethodTableMap = target.ReadPointer(address + (ulong)type.Fields[nameof(TypeDefToMethodTableMap)].Offset);
+        TypeRefToMethodTableMap = target.ReadPointer(address + (ulong)type.Fields[nameof(TypeRefToMethodTableMap)].Offset);
+    }
+
+    public TargetPointer Assembly { get; init; }
+    public uint Flags { get; init; }
+    public TargetPointer Base { get; init; }
+    public TargetPointer LoaderAllocator { get; init; }
+    public TargetPointer ThunkHeap { get; init; }
+
+    public TargetPointer FieldDefToDescMap { get; init; }
+    public TargetPointer ManifestModuleReferencesMap { get; init; }
+    public TargetPointer MemberRefToDescMap { get; init; }
+    public TargetPointer MethodDefToDescMap { get; init; }
+    public TargetPointer TypeDefToMethodTableMap { get; init; }
+    public TargetPointer TypeRefToMethodTableMap { get; init; }
+
+    private TargetPointer _metadataStart = TargetPointer.Null;
+    private ulong _metadataSize;
+    public TargetPointer GetLoadedMetadata(out ulong size)
+    {
+        if (Base != TargetPointer.Null && _metadataStart == TargetPointer.Null && _metadataSize == 0)
+        {
+            int peSignatureOffset = _target.Read<int>(Base + PEOffsets.DosStub.PESignatureOffset);
+            ulong headerOffset = Base + (ulong)peSignatureOffset;
+            ushort magic = _target.Read<ushort>(headerOffset + PEOffsets.PEHeader.Magic);
+            ulong clrHeaderOffset = magic == (ushort)PEMagic.PE32
+                ? PEOffsets.PEHeader.CLRRuntimeHeader32
+                : PEOffsets.PEHeader.CLRRuntimeHeader32Plus;
+            int corHeaderRva = _target.Read<int>(headerOffset + clrHeaderOffset);
+
+            // Read RVA and size of the metadata
+            ulong metadataDirectoryAddress = Base + (ulong)corHeaderRva + PEOffsets.CorHeader.Metadata;
+            _metadataStart = Base + (ulong)_target.Read<int>(metadataDirectoryAddress);
+            _metadataSize = (ulong)_target.Read<int>(metadataDirectoryAddress + sizeof(int));
+        }
+
+        size = _metadataSize;
+        return _metadataStart;
+    }
+
+    // https://learn.microsoft.com/windows/win32/debug/pe-format
+    private static class PEOffsets
+    {
+        private const int PESignatureSize = sizeof(int);
+        private const int CoffHeaderSize = 20;
+
+        public static class DosStub
+        {
+            public const int PESignatureOffset = 0x3c;
+        }
+
+        public static class PEHeader
```

Comment:
Perhaps `PEHeaderOffsets`?

Reply:
If not, we should have `Size` or `Offset` in the name of the constant. I assume there will be a case where it can be both, so in those cases a comment seems appropriate.